app/src/main/java/com/peterlaurence/trekme/core/map/data/dao/MapDownloadDaoImpl.kt

```diff
@@ -74,9 +74,8 @@ class MapDownloadDaoImpl(
             try {
                 val out = FileOutputStream(tileFile)
                 bitmap.compress(Bitmap.CompressFormat.JPEG, 90, out)
-                val tag = makeTag(request.source)
-                if (tag != null) {
-                    out.write(tag)
+                makeTag(request.source)?.let {
```

Comment:
`also` would be semantically a little better than `let`, since we don't care about the returned type.